catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/watcher/FilesWatcher.java

```diff
@@ -0,0 +1,124 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor.watcher;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.codice.ddf.platform.util.StandardThreadFactoryBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Periodically calls {@link FileWatcher#check()} on all watched {@link FileWatcher}s. */
+public class FilesWatcher {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FilesWatcher.class);
+
+  private static final long DEFAULT_POLL_TIME_SEC = 5;
+
+  private static final String CDM_FILE_CHECK_PERIOD_PROPERTY = "org.codice.ddf.cdm.fileCheckPeriod";
+
+  private ScheduledExecutorService executorService;
+
+  private final Map<File, FileWatcher> watchers;
+
+  private long executorPollTime;
+
+  /**
+   * Creates a new FileWatcher with the system property {@link #CDM_FILE_CHECK_PERIOD_PROPERTY} poll
+   * time or {@link #DEFAULT_POLL_TIME_SEC} if not available.
+   */
+  public FilesWatcher() {
+    watchers = new ConcurrentHashMap<>();
+    executorPollTime = getPollTimeOrDefault();
+    init();
+  }
+
+  /**
+   * Creates a new FileWatcher with the given polling interval.
+   *
+   * @param pollTime polling interval in seconds
+   */
+  public FilesWatcher(long pollTime) {
+    watchers = new ConcurrentHashMap<>();
+    executorPollTime = pollTime;
+    init();
+  }
+
+  /**
+   * Submits a {@link FileWatcher} whose {@link FileWatcher#check()} will be called periodically.
+   *
+   * @param fileWatcher to submit for checking
+   */
+  public void watch(FileWatcher fileWatcher) {
+    watchers.put(fileWatcher.getWatchedFile(), fileWatcher);
+  }
+
+  private void init() {
+    executorService =
+        Executors.newSingleThreadScheduledExecutor(
+            StandardThreadFactoryBuilder.newThreadFactory("cdmFileWatcher"));
+
+    executorService.scheduleAtFixedRate(this::checkFiles, 10, executorPollTime, TimeUnit.SECONDS);
+  }
+
+  /**
+   * Calls {@link FileWatcher#check()} on all registered file watchers. For anyone that returns
+   * true, the {@link FileWatcher} is removed from this {@code FilesWatcher}.
+   */
+  private void checkFiles() {
+    for (Entry<File, FileWatcher> entry : watchers.entrySet()) {
+      File watchedFile = entry.getKey();
+      FileWatcher watcher = entry.getValue();
+
+      if (watcher.check()) {
+        watchers.remove(watchedFile);
+      }
+    }
+  }
+
+  /** Must be called to clean up ExecutorService resources. */
+  public void destroy() {
```

Comment:
✏️ Since `destroy()` is `public`, do you think we should also make `init()` `public` and have clients call it to get things started?